File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    calc_cnt = 0
    while calc_cnt < 500:
        if num == 1:
            return calc_cnt
        if num % 2 == 0:
            num/=2
        else:
            num=(num*3)+1
        calc_cnt+=1
    return -1

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public long solution(long num) {
        int answer = 0;
        if(num == 1){
            return 0;
        }
        
        while(answer < 500){
            if(num == 1){
                return answer;
            }
            
            if(num % 2 == 0){
                num /= 2;
            }
            else{
                num *= 3;
                num += 1;
            }
            
            answer++;
        }
        return -1;
    }
}